### ISSUE TITLE
error: "unable to retrieve version information for virtualenv" (from pypi)

### ISSUE THREAD

**Issue by nikolay** [mjpizz/node-virtualenv#7]
```
$ ./node_modules/.bin/virtualenv-postinstall
(node:12121) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
Finding virtualenv *
Error: unable to retrieve version information for virtualenv
  at /Users/myuser/Projects/myproject/node_modules/virtualenv/lib/virtualenv.js:184:14
  at /Users/myuser/Projects/myproject/node_modules/pypi/src/pypi/client.coffee:15:17
  at /Users/myuser/Projects/myproject/node_modules/xmlrpc/lib/client.js:130:9
  at callback (/Users/myuser/Projects/myprojectt/node_modules/xmlrpc/lib/deserializer.js:30:7)
  at Deserializer.onDone (/Users/myuser/Projects/myproject/node_modules/xmlrpc/lib/deserializer.js:79:12)
  at emitNone (events.js:86:13)
  at SAXStream.emit (events.js:185:7)
  at Object.SAXStream._parser.onend (/Users/myuser/Projects/myproject/node_modules/xmlrpc/node_modules/sax/lib/sax.js:190:10)
  at emit (/Users/myuser/Projects/myproject/node_modules/xmlrpc/node_modules/sax/lib/sax.js:640:35)
  at end (/Users/myuser/Projects/myproject/node_modules/xmlrpc/node_modules/sax/lib/sax.js:683:5)
  at Object.SAXParser.end (/Users/myuser/Projects/myproject/node_modules/xmlrpc/node_modules/sax/lib/sax.js:154:24)
  at SAXStream.end (/Users/myuser/Projects/myproject/node_modules/xmlrpc/node_modules/sax/lib/sax.js:248:18)
  at IncomingMessage.onend (_stream_readable.js:512:10)
  at IncomingMessage.g (events.js:286:16)
  at emitNone (events.js:91:20)
  at IncomingMessage.emit (events.js:185:7)
  at endReadableNT (_stream_readable.js:973:12)
  at _combinedTickCallback (internal/process/next_tick.js:74:11)
  at process._tickCallback (internal/process/next_tick.js:98:9)
```

I have both Python 2 and Python 3 installed, virtualenv is installed under both, too.

```
$ python --version
Python 2.7.12
$ python3 --version
Python 3.5.2
$ virtualenv --version
1.11.4
$ which virtualenv
/usr/local/bin/virtualenv
```

**Comment by nikolay** [comment]
@mjpizz No feedback?

**Comment by mjpizz** [comment]
hey there @nikolay sorry - I was out of town and just got to this. Couple things:

1. This looks very similar to #6, which was rooted in a [pypi issue](https://github.com/lukaszb/pypi/pull/3). It looks like @lukaszb merged the fix but hasn't published an updated pypi version to npmjs.org - I just messaged him on that PR to see what's up.
2. Your Python and virtualenv versions shouldn't affect this module, since it uses a downloaded version of virtualenv rather than relying on the system.

While you're waiting for the pypi module to be republished, you should be able to add this workaround in your package.json dependencies:

```
    "pypi": "git://github.com/lukaszb/pypi.git"
```

That should force your app to pick up the latest version of pypi from Github rather than npmjs.org in the meantime.

**Comment by nikolay** [comment]
@mjpizz Okay, using the `pypi` module from GitHub vs PyPI solved the issue but, honestly, the `pypi` package is just a few lines of an XMLRPC wrapper - do you really need it as a dependency and a constant source of issues?

**Comment by nikolay** [comment]
@mjpizz You also seem to drag in the whole CoffeeScript as a dependency along with the `pypi` package!

**Comment by davidar** [comment]
@nikolay lukaszb/pypi#4 would remove the runtime coffeescript dependency

**Comment by nikolay** [comment]
@davidar Thanks!

**Comment by mjpizz** [comment]
@nikolay fair point on pypi+coffeescript dependency. I'm pretty busy next couple weeks, but if you wanted to put together a pull request I could probably carve out some time to review and publish. Let me know!

**Comment by mjpizz** [comment]
Haven't heard back from @lukaszb yet about publishing the [latest version of pypi](https://github.com/lukaszb/pypi/pull/3#issuecomment-237267686) to npm, so I went ahead and vendored `pypi` as precompiled JS. This is issue should be resolved for you by using `virtualenv` 0.3.1, let me know if you have any other issues with it!

**Comment by nikolay** [comment]
Thank you very much!
